.github/workflows/macos.yml

```diff
@@ -0,0 +1,28 @@
+name: MacOs
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: macos-latest
+    strategy:
+      matrix:
+        mpi: [ ON, OFF ] 
+        compiler: [gcc]
```

Comment:
For me it would seem more logical to test with clang as that is the default compiler on Mac.